src/go/monitor/main.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"flag"
+	"log"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// serviceMetrics holds metrics for a service
+type serviceMetrics struct {
+	VisitTotal prometheus.Counter
+}
+
+// newServiceMetrics registers metrics
+func newServiceMetrics() serviceMetrics {
+	return serviceMetrics{
+		VisitTotal: promauto.NewCounter(prometheus.CounterOpts{
+			Name: "myapp_visit_total",
+			Help: "The total number of visits",
+		}),
+	}
+}
+
+// Service of the app
+type Service struct {
+	metrics serviceMetrics
+}
+
+// NewService returns a new service.
+func NewService() Service {
+	return Service{
+		metrics: newServiceMetrics(),
+	}
+}
+
+// Handler return a HTTP handler for the service.
+func (s Service) Handler() http.Handler {
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", promhttp.Handler())
```

Comment:
Can we add a simple root page to show that the service works? Nothing fancy a "OK" or a link to the metrics page would be better then a 404 😃